src/api/rewardsHistory.ts

```diff
@@ -15,36 +15,50 @@ async function extractVaultUserRewards(
-        variables: vars_getRewards,
-        onSuccess: function (value: Response): Response {
-            return value;
-        },
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        onError: function (reason: any): PromiseLike<never> {
-            throw new Error(`Failed to get rewards from Stakewise: ${reason}`);
-        },
-    });
+    try {
+        const responseRewards = await connector.graphqlRequest({
+            type: 'api',
+            op: 'UserRewards',
+            query: `query UserRewards($user: String!, $vaultAddress: String!, $dateFrom: DateAsTimestamp!) { userRewards(user: $user, vaultAddress: $vaultAddress, dateFrom: $dateFrom) { date, sumRewards, }}`,
+            variables: vars_getRewards,
+            onSuccess: function (value: Response): Response {
+                return value;
+            },
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            onError: function (reason: any): PromiseLike<never> {
+                throw new Error(`Failed to get rewards from Stakewise: ${reason}`);
+            },
+        });
+        if (!responseRewards.ok) {
+            throw new Error('Invalid response from Stakewise');
+        }
+        const rewardsData = await responseRewards.json();
+
+        if (rewardsData.errors && rewardsData.errors.length) {
+            throw new Error(rewardsData.errors[0].message);
+        }
 
-    const rewardsData = await responseRewards.json();
-    const dataPoints: RewardsDataPoint[] = [];
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    rewardsData.data.userRewards.forEach((reward: any) => {
-        const when: Date = new Date(parseInt(reward.date) * 1000);
-        if (when <= dateTo) {
-            const sumRewards: string = reward.sumRewards;
-            dataPoints.push({
-                when: when,
-                amount: BigInt(sumRewards),
-                vault: vault,
-            });
+        if (!rewardsData.data || !rewardsData.data.userRewards || rewardsData.data.userRewards.length === 0) {
+            throw new Error('Rewards data is missing or incomplete');
         }
-    });
+        const dataPoints: RewardsDataPoint[] = [];
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        rewardsData.data.userRewards.forEach((reward: any) => {
+            const when: Date = new Date(parseInt(reward.date) * 1000);
+            if (when <= dateTo) {
+                const sumRewards: string = reward.sumRewards;
+                dataPoints.push({
+                    when: when,
+                    amount: BigInt(sumRewards),
+                    vault: vault,
+                });
+            }
+        });
 
-    return dataPoints;
+        return dataPoints;
+    } catch (error) {
+        throw new Error(`Error retrieving rewards history data: ${error instanceof Error ? error.message : error}`);
```

Comment:
I think stack trace will be lost here.
One idea to keep it would be just changing the message in the errors throw before
i.e. And just rethrowing the error itself.
```
throw new Error('Error retrieving rewards history data: Rewards data is missing or incomplete');
```
instead of 
```
throw new Error('Rewards data is missing or incomplete');
```

or using some manipulation via Error.stack, or Error.captureStackTrace

consider adding it